Made it possible to set the options for the request

blaues0cke opened the pull request:
I need this since I was trying to get my GitHub user data (https://api.github.com/users/blaues0cke) and `getJSON` failed without an error. Figured out that GitHub responds with

```
'Request forbidden by administrative rules. Please make sure your request has a User-Agent header (http://developer.github.com/v3/#user-agent-required). Check https://developer.github.com for other possible causes.\n' } Request forbidden by administrative rules. Please make sure your request has a User-Agent header (http://developer.github.com/v3/#user-agent-required). Check https://developer.github.com for other possible causes.` so I had to make it possible to pass the options to set a `User-Agent`
```

Example code:

```
var apiUrl = 'https://api.github.com/users/blaues0cke';

getJSON(apiUrl, { 'headers': { 'User-Agent': 'request' } }, function(error, response) {

});
```

(Dont know if this is compatible with the api from `jsonp`.

Blauesocke committed:
Made it possible to set the options for the request (needed this since api calls to github require a user agent)

zeke commented on lib/node.js:
```diff
@@ -2,12 +2,23 @@ var request = require("request");
+  var defaultOptions = { url: url, json: true };
+
+  if (options) {
+    defaultOptions = Object.assign(defaultOptions, options);
```
`Object.assign` is only available since node 4. I'm fine with this, but please add a stanza to package.json to make node support explicit:

```json
{
  "engines": {
    "node": ">=4"
  }
}
```

zeke reviewed (changes requested):
I'm happy to land this feature. Can you add some tests for this new behavior?

zeke commented:
bump 👋

blaues0cke commented:
Sorry, missed the notification. Will do this in the next days.

blaues0cke committed:
Added a new test for the optional options parameter (https://github.com/zeke/get-json/pull/1#discussion_r84623788)

blaues0cke commented:
Had to replace the old testing framework with `tape`. I also fixed the existing tests since the used api is no longer available. And of course I added a new test for the feature I added. See the commit messages for more detailed informations.

```
thomas.kekeisen@tkwin ~/htdocs/get-json
$ npm test

> get-json@0.0.3 test A:\xampp\htdocs\get-json
> node test

TAP version 13
# get last libraries from cdnjs.com
ok 1 undefined
ok 2 should be truthy
ok 3 should be equal
# get repositories from GitHub
ok 4 undefined
ok 5 should be truthy
ok 6 should be truthy
ok 7 should be equal

1..7
# tests 7
# pass  7

# ok
```

zeke commented:
Looks good. Can you update `.travis.yml` to test on node 4 and 6?

ljharb commented:
I've rebased this, and dropped the now-obviated test changes, and the engines changes (if we need Object.assign functionality, we'll use the object.assign package, but I don't think we do). Since get-json now uses phin instead of request, there's merge conflicts that'll need resolving.
